datajs/src/file-base.js

```diff
@@ -140,13 +142,36 @@ export class FileInterface extends File {
     return this.descriptor.name
   }
 
-  async hash() {
-    const text = await this.descriptor.text()
-    return crypto.createHash('md5').update(text).digest('hex')
-  }
-
-  async hashSha256() {
-    const text = await this.descriptor.text()
-    return crypto.createHash('sha256').update(text).digest('hex')
+  generateHash(hashType, cbProgress) {
+    return new Promise((resolve, reject) => {
+      let newHash =  hashType === "md5" ? CryptoJS.algo.MD5.create() : CryptoJS.algo.SHA256.create();
```

Comment:
You could achieve this also in previous `hash()` implementation.  `crypto.createHash('md5').update(text).digest('hex')` this is for MD5 and for `sha256` 

`crypto.createHash('sha256').update(text).digest('hex')`